src/main/java/ch/njol/skript/expressions/ExprFormatTime.java

```diff
@@ -51,35 +59,50 @@
 		Skript.registerExpression(ExprFormatTime.class, String.class, ExpressionType.PROPERTY, "%dates% formatted [human-readable] [(with|as) %-string%]");
 	}
 
-	@SuppressWarnings("null")
+	@SuppressWarnings("NotNullFieldNotInitialized")
 	private SimpleDateFormat format;
 
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	private Expression<? extends String> customFormat;
+
 	@Override
-	@SuppressWarnings("null")
+	@SuppressWarnings({"null", "unchecked"})
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		setExpr((Expression<? extends Date>) exprs[0]);
-		if (exprs[1] != null) {
-			if (!(exprs[1] instanceof Literal)) {
-				VariableString str = (VariableString) exprs[1];
-				if (!str.isSimple()) {
-					Skript.error("Date format must not contain variables!");
-					return false;
+		customFormat = (Expression<? extends String>) exprs[1];
+
+		if (customFormat != null) {
+			if (!(customFormat instanceof Literal) && customFormat instanceof VariableString) {
+				VariableString str = (VariableString) customFormat;
+				if (str.isSimple()) {
+					try {
+						format = new SimpleDateFormat(customFormat.getSingle(null));
+					} catch (Exception e) {
+						Skript.error("Incorrect date format used: " + e.getMessage());
+						return false;
+					}
 				}
 			}
-			format = new SimpleDateFormat((String) exprs[1].getSingle(null));
-		} else {
-			format = new SimpleDateFormat(defaultFormat);
 		}
-		
+
 		return true;
 	}
 
-
 	@Override
 	protected String[] get(Event e, Date[] source) {
 		return get(source, new Getter<String, Date>() {
 			@Override
 			public String get(Date date) {
+				if (customFormat != null) {
+					try {
+						format = new SimpleDateFormat(customFormat.getSingle(e));
+					} catch (Exception ex) {
+//						Skript.warning("Incorrect date format used: " + ex.getMessage());
+						format = new SimpleDateFormat(defaultFormat);
```

Comment:
based on the discussion before, this would return null I think since this means that the user-provided date was invalid

Reply:
I forgot to mention this here as I was chatting with TP in DMs, I have made it like this (with a note in the description) to try to keep the result not null to avoid the possibility of causing some unexpected code breaks if the format is incorrect and because there are no warnings to warn the user about such case (avoiding console spam), in this case they will at least expect some output instead of having <none> (tho both cases could be confusing but the description note should clarify)

However, if you think this behavior is not the best I can make it return null as we spoke earlier

Reply:
I dunno, I definitely see where you guys are coming from but I think that returning null makes more sense than returning a different result. This is because, to me, it's a lot easier to notice something isn't right if I see `<none>` in my output vs an alternative date format which I might overlook at a glance. Plus, returning null allows a scripter to easily check if the format was correct:
```
set {_formatted now} to now formatted as {_super cool custom format}
if {_formatted now} is not set: # the format passed was invalid
```

Reply:
IMO it should return null, not some random value

Reply:
Makes more sense in your case, will do that